integration-tests/plugins/__tests__/price-lists/store/get-products.ts

```diff
@@ -0,0 +1,339 @@
+    await db.teardown()
+  })
+
+  it("should list product and prices from price-list", async () => {
+    const [{ id: priceListId }] = await pricingModuleService.createPriceLists([
+      {
+        title: "test price list",
+        description: "test",
+        status: PriceListStatus.ACTIVE,
+        type: PriceListType.SALE,
+        prices: [
+          {
+            amount: 2500,
+            currency_code: "usd",
+            price_set_id: priceSetId,
+          },
+        ],
+      },
+    ])
+
+    const api = useApi() as any
+
+    let response = await api.get(
+      `/store/products/${product.id}?currency_code=usd`
+    )
+
+    expect(response.status).toEqual(200)
+    expect(response.data.product.variants[0].prices).toHaveLength(2)
+    expect(response.data.product.variants[0].prices).toEqual([
+      expect.objectContaining({
+        currency_code: "usd",
+        amount: 3000,
+        min_quantity: null,
+        max_quantity: null,
+        price_list_id: null,
+      }),
+      expect.objectContaining({
+        currency_code: "usd",
+        amount: 2500,
+        min_quantity: null,
+        max_quantity: null,
+        price_list_id: priceListId,
+      }),
+    ])
+    expect(response.data.product.variants[0]).toEqual(
+      expect.objectContaining({
+        original_price: 3000,
+        calculated_price: 2500,
+        calculated_price_type: "sale",
+      })
+    )
+  })
+
+  it("should list product and prices from price-list created in the api", async () => {
+    const api = useApi()! as AxiosInstance
+
+    const priceListResponse = await api.post(
+      `/admin/price-lists`,
+      {
+        name: "test price list",
+        description: "test",
+        status: PriceListStatus.ACTIVE,
+        type: PriceListType.SALE,
+        prices: [
+          {
+            amount: 2500,
+            currency_code: "usd",
+            variant_id: variant.id,
+          },
+        ],
+      },
+      adminHeaders
+    )
+
+    let response = await api.get(
+      `/store/products/${product.id}?currency_code=usd`
+    )
+
+    expect(response.status).toEqual(200)
+    expect(response.data.product.variants[0].prices).toHaveLength(2)
+    expect(response.data.product.variants[0].prices).toEqual([
+      expect.objectContaining({
+        currency_code: "usd",
+        amount: 3000,
+        min_quantity: null,
+        max_quantity: null,
+        price_list_id: null,
+      }),
+      expect.objectContaining({
+        currency_code: "usd",
+        amount: 2500,
+        min_quantity: null,
+        max_quantity: null,
+        price_list_id: priceListResponse.data.price_list.id,
+      }),
+    ])
+    expect(response.data.product.variants[0]).toEqual(
+      expect.objectContaining({
+        original_price: 3000,
+        calculated_price: 2500,
+        calculated_price_type: "sale",
+      })
+    )
+  })
+
+  it("should not list prices from price-list with customer groups if not logged in", async () => {
```

Comment:
not sure I understand your comment here?

don't mind merging the two tests, I just liked having the two different test cases in different tests

Reply:
The first one is testing when creating through the module, in oposition to the second one which stipulate through the api. here we should only test through the API since through the module is tested in the integration tests of the module. So I don't see why we would have a specific test to tests through the API since it is the purpose of those integration tests by nature as it test the integration in the core so through the end point

Reply:
ah, you are commenting on a different test ("should list product and prices from price-list") in the same file that's why I got confused, we can remove the module test no issue